### ISSUE TITLE
Blank page after login

### ISSUE THREAD

**Issue by franko108** [LDAPAccountManager/lam#106]
Hello,
I've just installed ldap account manager.  
I did initial setup within "Edit Server Profiles" section. Connection goes via ldap on port 389 with TLS.   
Testing in  a shell with `ldapwhoami -H ldap://my-domain -x -v -ZZ` response is proper, openldap with TLS looks fine.   
The same SSL (provided by letsencrypt)  is used 
On the _List of valid users_ there is admin, the same as asking `slapcat` command in a shell. 
When I try to login with the admin I got a blank page.  
In syslog there is a message `successfully logged in.`  
Also:  
`Sep  2 09:13:47 spree-ldap1 php: LDAP Account Manager (7v71av5ispi29lalquc9ovdd0i - 151.252.225.9 - cn=admin,dc=my-domain) - WARNING: Unable to set locale, check if 'locale -a' returns en_GB.utf8
`  
(last looks as a warning, not a fatal error).  
The next line says (apache output):  
```
spree-ldap1.sonnen.de:443 151.252.225.9 - - [02/Sep/2020:08:52:37 +0200] "GET /lam/templates/main.php HTTP/1.1" 500 946 "https://my-domain/lam/templates/login.php?configSaveOk=1&configSaveFile=lam" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:79.0) Gecko/20100101 Firefox/79.0" 
```
So, error 500. Enabling in php.ini `error_reporting = E_ALL` doesn't give any report on the screen.  
Tried to disable TLS, same result.  
On the browser - developer tools, there is a message related to error 500 (URI templates/main.php):
```
	receiveMessage/<
resource:///actors/BrowserTabChild.jsm:102:26
wrapHandlingUserInput
resource://gre/modules/E10SUtils.jsm:1009:7
receiveMessage
resource:///actors/BrowserTabChild.jsm:99:21
```

There was some additional error in apache log that I can't repeat:  
```
[Wed Sep 02 08:35:38.894596 2020] [php7:error] [pid 149660] [client 151.252.225.9:63984] PHP Fatal error:  Uncaught Error: Class 'locking389ds' not found in /usr/share/ldap-account-manager/templates/main.php:57\nStack trace:\n#0 {main}\n  thrown in /usr/share/ldap-account-manager/templates/main.php on line 57, referer: https://my-domain/lam/templates/login.php?configSaveOk=1&configSaveFile=lam
[Wed Sep 02 08:39:39.731991 2020] [php7:warn] [pid 149814] [client 151.252.225.9:63998] PHP Warning:  ldap_start_tls(): Unable to start TLS: Connect error in /usr/share/ldap-account-manager/lib/account.inc on line 733, referer: https://spree-ldap1.sonnen.de/lam/templates/login.php?configSaveOk=1&configSaveFile=lam
```

(BTW, a FQDN has been actually used instead of _my-domain_)  
Apache config is default as provided by ldap account manager.

- PHP version: 7.4.3
- Apache version 2.4.41 (Ubuntu)

---
Also, is there anyway or need to disable CSRF checking as I want to have haproxy on different domain in front of this apache?

Thanks in advance.

**Comment by gruberroland** [comment]
Please use LAM 7.2 instead of 7.3.RC1. This should solve the issue with the PHP fatal error. Will be fixed in final 7.3.

For TLS you need to import your certificate in LAM's main configuration:

https://www.ldap-account-manager.org/static/doc/manual/ch03.html#idm713

**Comment by franko108** [comment]
Hi @gruberroland - version 7.2 works, thanks a lot.  
I did download from official web version 7.3.RC1, well I guess it is better to put a stable version there? :-)   

However, there was an error (actually warning as I have now displaying verbose error ):
```
Deprecated: Array and string offset access syntax with curly braces is deprecated in /usr/share/ldap-account-manager/templates/3rdParty/pla/lib/functions.php on line 1449
```
On that line is:
`'$' == $rdn{ strlen($rdn) - 1 })`  
I think it should be square brackets instead of curly bracket:
`$' == $rdn[strlen($rdn) - 1 ])`
Then there is no error on the same page.

BTW, I didn't import TLS cert, but it looks good so far (I just put in apache vhost, the config path to cert and key, maybe is this enough?)

This software looks good, I think it will be useful for our need.  
Thank you very much for help, all the best.

**Comment by gruberroland** [comment]
Great that it works now. :)
The brackets issue is already fixed and will be published with final 7.3 in a few days.
Could be that the Apache setting is already fine. Unfortunately, not all systems use the same configs. Therefore, we also provide the cert import inside LAM.
